client/components/prompt-input/PromptInputForm.tsx

```diff
@@ -44,17 +47,37 @@ const GenerateButton = styled(Button)<ButtonProps>(({ theme }) => ({
   textTransform: 'none',
 }));
 
+const UserInputCheckbox = styled(Checkbox)(({ theme }) => ({
+  color: theme.palette.primary.main,
+}));
+
 const PromptInputForm = () => {
-  const [inputValue, setInputValue] = useState('');
-  const [isChecked, setIsChecked] = useState(false);
+  // Here we utilize the useForm hook's properties to manage form state while also initializing default values and resolving our validation schema.
+  // The control object is used by react-hook-form's Controller to link our UserInputBar and UserInputCheckbox components to the form's state, handling any necessary state updates or validation checks.
+  const {
+    handleSubmit,
+    reset,
+    control,
+    watch,
+    formState: { errors, isSubmitted },
+  } = useForm({
+    resolver: zodResolver(userInputValidationSchema),
+    defaultValues: {
+      input: '',
+      isChecked: JSON.parse(localStorage.getItem('isChecked') || 'false'),
```

Comment:
Could you change this to

`isChecked: Boolean(localStorage.getItem('isChecked')) || false,`

You can add this to a followup pull request